rar.go

```diff
@@ -37,6 +38,23 @@ func Unrar(source, destination string) error {
 		} else if err != nil {
 			return err
 		}
+		
+		pathComponents := strings.Split(header.Name, "/")
+
+		for pi, path := range pathComponents {
+			// the last component of the path will be the file
+			// so ignore it, since we only want to create folders
+			if pi == len(pathComponents)-1 {
+				continue
+			}
+
+			// check to see if the path exists already
+			if stat, err := os.Stat(destination + path); err != nil || !stat.IsDir() {
+				// make the directory
+				mkdir(destination + path)
```

Comment:
This function will make all parent folders necessary to make the folder at `destination+path` so it should only need be called once.